wled00/FX.cpp

```diff
@@ -6443,8 +6434,8 @@ uint16_t mode_2DWaverly(void) {
   const int cols = SEGMENT.virtualWidth();
   const int rows = SEGMENT.virtualHeight();
 
-  um_data_t *um_data = getAudioData();
-  float   volumeSmth  = *(float*)   um_data->u_data[0];
+  um_data_t *um_data = getAudioData(SEGMENT.soundSim);
```

Comment:
Why pass a variable that was already visible inside the method?

Reply:
I was thinking about visibility change in the future.

Reply:
Passing by reference may save on code size, just a guess.

Reply:
There are 29 calls to `getAudioData()`. While it is true that `SEGMENT.soundSim` is globally accessible, it is only due to the fact that, if hidden behind function calls, the code would increase tremendously (done that).

> Passing by reference may save on code size

`soundSim` is a bitfield so passing by reference is not possible.